src/storage/mvcc/reader.rs

```diff
@@ -273,6 +273,48 @@ impl<'a> MvccReader<'a> {
         Ok(())
     }
 
+    pub fn seek_first(&mut self, ts: u64) -> Result<Option<Key>> {
+        assert!(self.scan_mode.is_some());
+        self.create_write_cursor()?;
+        self.create_lock_cursor()?;
+
+        let mut w_cur = self.write_cursor.as_mut().unwrap();
+        let mut l_cur = self.lock_cursor.as_mut().unwrap();
+        let (mut w_ok, mut l_ok) = (w_cur.seek_to_first(&mut self.statistics.write),
+                                    l_cur.seek_to_first(&mut self.statistics.lock));
+
+        loop {
+            let (mut w_key, mut l_key) = (None, None);
+            if w_ok {
+                w_key = Some(w_cur.key().to_vec());
```

Comment:
Could you make the `w_key=Some(Key::from_encoded(wk).truncate_ts())` directly to avould  `w_key.clone()` ?